### PR TITLE
Separate execution's result

### PR DESCRIPTION
**Pull request by NicolasMahe**
This PR separates the execution's result to a dedicated SDK and API.

closes https://github.com/mesg-foundation/engine/issues/1542

### CODE REVIEW

File: orchestrator/orchestrator.go [hunk 1]
```diff
@@ -30,30 +32,36 @@ func New(event EventSDK, execution ExecutionSDK, process ProcessSDK, runner Runn
...
 	s.eventStream = s.event.GetStream(nil)
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	executionStream, errC, err := s.execution.Stream(ctx, &api.StreamExecutionRequest{
-		Filter: &api.StreamExecutionRequest_Filter{
-			Statuses: []execution.Status{execution.Status_Completed},
-		},
+	resultStream, errC, err := s.result.Stream(ctx, &api.StreamResultRequest{
+		//TODO: filter out result with error or catch them and put the error on the errChan
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]: TODO

File: orchestrator/orchestrator.go [hunk 2]
```diff
@@ -30,30 +32,36 @@ func New(event EventSDK, execution ExecutionSDK, process ProcessSDK, runner Runn
...
-	s.executionStream = executionStream
+	s.resultStream = resultStream
 	for {
 		select {
 		case event := <-s.eventStream.C:
-			go s.execute(s.eventFilter(event), nil, event, event.Data)
-		case execution := <-s.executionStream:
-			go s.execute(s.resultFilter(execution), execution, nil, execution.Outputs)
-			go s.execute(s.dependencyFilter(execution), execution, nil, execution.Outputs)
+			// TODO: are we sure the function in param (s.eventFilter) is also call in the go func?
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]: @krhubert can you answer this comment question

File: protobuf/types/execution.proto [hunk 3]
```diff
@@ -76,19 +51,9 @@ message Execution {
...
-
-  // error message of a failed execution.
-  string error = 9 [
-    (gogoproto.moretags) = 'hash:"-"'
-  ];
-
   // tags are optionally associated with execution by the user.
   repeated string tags = 10 [
-    (gogoproto.moretags) = 'hash:"name:10" validate:"dive,printascii"'
+    (gogoproto.moretags) = 'hash:"name:10"'
```
Comment (NicolasMahe) [inline protobuf/types/execution.proto]:
```suggestion
    (gogoproto.moretags) = 'hash:"name:10" validate:"dive,printascii"'
```

File: protobuf/types/result.proto [hunk 4]
```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
```
Comment (NicolasMahe) [inline protobuf/types/result.proto]: add validation

File: protobuf/api/result.proto [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  // Stream returns a stream of executions that satisfy criteria
+  // specified in a request.
+  rpc Stream(StreamResultRequest) returns (stream types.Result) {}
+}
+
+// CreateResultRequest defines request to create a single result.
+message CreateResultRequest {
+  // requestHash is hash of the associated execution.
+  bytes requestHash = 1 [
```
Comment (krhubert) [inline protobuf/api/result.proto]:
```suggestion
  bytes executionHash = 1 [
```

We still have `Execution` type, not `Request` what do you thing?

File: scripts/build-mocks.sh [hunk 6]
```diff
@@ -2,6 +2,7 @@
 
 # generate mocks
 mockery -name ExecutionSDK -dir ./orchestrator -output ./orchestrator/mocks
+mockery -name ResultSDK -dir ./orchestrator -output ./orchestrator/mocks
```
Comment (krhubert) [inline scripts/build-mocks.sh]: this mock is not used anywhere so generation is not needed.

File: protobuf/api/result.proto (hunk 5 above)
Comment (NicolasMahe) [reply protobuf/api/result.proto]: good catch

File: orchestrator/orchestrator.go (hunk 1 above)
Comment (NicolasMahe) [reply orchestrator/orchestrator.go]: done

File: orchestrator/orchestrator.go (hunk 2 above)
Comment (NicolasMahe) [reply orchestrator/orchestrator.go]: the param function are not called in the go routine

File: protobuf/types/result.proto (hunk 4 above)
Comment (NicolasMahe) [reply protobuf/types/result.proto]: will be done in another PR

File: orchestrator/orchestrator_test.go [hunk 7]
```diff
@@ -1,271 +1,314 @@
 package orchestrator
 
-import (
-	"fmt"
-	"testing"
+// import (
```
Comment (NicolasMahe) [inline orchestrator/orchestrator_test.go]: put back test

File: scripts/build-mocks.sh (hunk 6 above)
Comment (NicolasMahe) [reply scripts/build-mocks.sh]: i updated the test and now it's used in the test

**Comment by NicolasMahe** [comment]
closing this PR because it's too old

**Pull request closed without merging**